compiler/AST/ParamForLoop.cpp

```diff
@@ -61,7 +61,35 @@ BlockStmt* ParamForLoop::buildParamForLoop(VarSymbol* indexVar,
     stride = new SymExpr(new_IntSymbol(1));
   }
 
-  if (call && call->isNamed("chpl_build_bounded_range"))
+  if (call && call->isNamed("#"))
+  {
+    Expr* count = new CallExpr("chpl_compute_count_param_loop", call->get(2)->remove());
+    call        = toCallExpr(call->get(1));
+    if (call && call->isNamed("chpl_build_low_bounded_range"))
+    {
+      low  = call->get(1)->remove();
+      high = new CallExpr("chpl_high_bound_count_for_param_loop", low->copy(), count);
+    }
+    else if (call && call->isNamed("chpl_build_high_bounded_range"))
+    {
+      high = call->get(1)->remove();
+      low  = new CallExpr("chpl_low_bound_count_for_param_loop", high->copy(), count);
+    }
+    else if (call && call->isNamed("chpl_build_bounded_range"))
+    {
+      Expr* temp_low = call->get(1);
+      Expr* temp_high = call->get(2);
```

Comment:
I'm trying to reason about why the previous cases used `->remove()` when getting their arguments while this case does not.  Could it, for symmetry?

Reply:
Sir, I tried to use it but the code was giving internal error on their usage. I tried looking up for an explanation but couldn't find anything satisfactory.

Reply:
Sorry, sir. I got my mistake. It should be get(1) for both as after I have removed temp_low, Temp_high gets moved to position 1. This is why it was giving error. I have corrected it.